neurox/interpretation/iou_probe.py

```diff
@@ -0,0 +1,31 @@
+from sklearn.metrics import average_precision_score
```

Comment:
Can we add some module level docs like:

"""Module for IoU method to rank neurons

This module implements the Intersection over Union method to rank neurons based on their activation values and the true class.

.. seealso::
      <citation>
"""